office/libreoffice/Portfile

```diff
@@ -0,0 +1,315 @@
+# Try to keep this in sync with the Python portgroup.
+set python_version 38
+depends_run-append \
+    port:python${python_version} \
+    port:py${python_version}-lxml
+
+if {${os.major} < 14} {
+    pre-fetch {
+        ui_error "${name} @${version} requires OS X 10.10 or later."
+        return -code error "incompatible OS X version"
+    }
+}
+
+patchfiles  pdfread-cxx-undefined.patch \
+            system-python.patch \
+            unpack-sources-fix-for-bsd-find.patch
+
+set product_name LibreOffice
+
+pre-configure {
+    system "cd ${worksrcpath} && env NOCONFIGURE=1 ./autogen.sh"
+}
+
+use_xcode           yes
+# CMS_NO_REGISTER_KEYWORD required for C++17 or newer compiler.
+configure.env-append  \
+    "LCMS2_CFLAGS=-I${prefix}/include -DCMS_NO_REGISTER_KEYWORD=1"
+# Most arguments are from
+# https://wiki.documentfoundation.org/LibreOffice_Vanilla_for_Mac#LibreOffice_Vanilla
+configure.args-append  \
+    --disable-breakpad \
+    --disable-bundle-mariadb \
+    --disable-ccache \
+    --disable-coinmp \
+    --disable-epm \
+    --disable-ext-ct2n \
+    --disable-ext-numbertext \
+    --disable-fetch-external \
+    --disable-firebird-sdbc \
+    --disable-ldap \
+    --disable-libnumbertext \
+    --disable-lotuswordpro \
+    --disable-lpsolve \
+    --disable-neon \
+    --disable-odk \
+    --disable-online-update \
+    --disable-pdfimport \
+    --disable-pdfium \
+    --disable-postgresql-sdbc \
+    --disable-qrcodegen \
+    --disable-report-builder \
+    --disable-vlc \
+    --enable-bogus-pkg-config \
+    --enable-cups \
+    --enable-extension-integration \
+    --enable-mpl-subset \
+    --enable-python=system \
+    --enable-readonly-installset \
+    --enable-release-build \
+    --enable-symbols \
+    --with-external-tar=${distpath} \
+    --with-myspell-dicts \
+    --with-package-version=${full_version} \
+    --with-parallelism=${build.jobs} \
+    --with-product-name=${product_name} \
+    --with-system-epoxy \
+    --with-system-gpgmepp \
+    --with-system-headers \
+    --with-system-hunspell \
+    --with-system-libebook \
+    --with-system-jars \
+    --with-system-libs \
+    --with-theme=colibre \
+    --with-tls=nss \
+    --with-vendor=MacPorts \
+    --without-doxygen \
+    --without-fonts \
+    --without-java \
+    --without-package-format \
+    --without-system-dicts \
+    --without-system-libabw \
+    --without-system-libcmis \
+    --without-system-libebook \
+    --without-system-libepubgen \
+    --without-system-libexttextcat \
+    --without-system-libmwaw \
+    --without-system-libqxp \
+    --without-system-libstaroffice \
+    --without-system-libzmf \
+    --without-system-mdds \
+    --without-system-mythes \
+    --without-system-orcus \
+    --without-system-sane \
+    --without-system-xmlsec \
+    \"--with-lang=bg br ca ca-valencia cy cs da de el en-US en-GB es et eu fi fr ga gd gl hr hu id is it ja km lt lv nb nl nn pl pt pt-BR ro ru sk sl sv ta tr uk zh-CN zh-TW\"
```

Comment:
Noting an issue and I do not know the cause yet:

On my system when I build with all these languages, the UI is in English as expected, but macOS' native dialog to open a file appears in Japanese (File -> Open or ⌘+O). The list of languages is right from LibreOffice's wiki. When I build with just 'en-GB en-US', the dialog appears in English. I imagine this affects other native dialogs. It does not affect dialog windows coming from UNO, nor any other UI.

Reply:
What you have set in Input Sources should not matter. What you have set in Language & Region might matter.

Reply:
And also the inconsistency between the app's localization on macOS dialogs' localization choices could be exactly the issue that was fixed in gettext 0.20 (which is not yet in MacPorts though there is a PR in progress).

Reply:
Checked with the official build and it doesn't show this issue. So it may be gettext.

Reply:
Still getting this issue on Big Sur.

Reply:
Still happens with 7.0.4.1.